src/Libraries/Microsoft.Extensions.AI.Abstractions/SpeechToText/SpeechToTextResponseUpdateExtensions.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Provides extension methods for working with <see cref="SpeechToTextResponseUpdate"/> instances.
+/// </summary>
+[Experimental("MEAI001")]
+public static class SpeechToTextResponseUpdateExtensions
+{
+    /// <summary>Combines <see cref="SpeechToTextResponseUpdate"/> instances into a single <see cref="SpeechToTextResponse"/>.</summary>
+    /// <param name="updates">The updates to be combined.</param>
+    /// <returns>The combined <see cref="SpeechToTextResponse"/>.</returns>
+    public static SpeechToTextResponse ToSpeechToTextResponse(
+        this IEnumerable<SpeechToTextResponseUpdate> updates)
+    {
+        _ = Throw.IfNull(updates);
+
+        SpeechToTextResponse response = new();
+        List<AIContent> contents = [];
+        string? responseId = null;
+        string? modelId = null;
+        object? rawRepresentation = null;
+        AdditionalPropertiesDictionary? additionalProperties = null;
+
+        foreach (var update in updates)
+        {
+            ProcessUpdate(update, contents, ref responseId, ref modelId, ref rawRepresentation, ref additionalProperties);
+        }
+
+        ChatResponseExtensions.CoalesceTextContent(contents);
+
+        response.Contents = contents;
+        response.ResponseId = responseId;
+        response.ModelId = modelId;
+        response.RawRepresentation = rawRepresentation;
+        response.AdditionalProperties = additionalProperties;
```

Comment:
Do StartTime and EndTime not come as part of streaming?

Reply:
They do, I added the missing fields as well as the missing UT ensuring the extension behavior.